Kernel/PreviousMode.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+
+namespace Kernel {
+
+enum class PreviousMode : u8 {
```

Comment:
`Space` sounds better to be than `Mode` in this context. Not sure if this enum should be just named `Space` and variable names should contain contextual information whether it previous or next (i.e. `Space`: `KernelSpace`, `UserSpace` or even `Kernel`, `User`)?

Reply:
I renamed the enum to ExecutionMode, as I thought that Space would be to similar to the AddressSpace.h class that we already have.